Issue #5066 - Abductor silencer now properly silences radios temporarily.

PidgeyThePirate opened the pull request:
## What Does This PR Do
Fixes #5066 - The silencer now does an emp_act on all the target's radios as well as turning broadcasting off, knocking them out for 10-20 seconds.

## Why It's Good For The Game
The silencer doesn't seem to have ever worked. Now it does.

## Changelog
:cl:
fix: The abductor's silencer now actually silences radios for 10-20 seconds.
/:cl:

Fox-McCloud commented:
This is papering over a bug and is absolutely the worst kind of bugfixing.

It's like when species is `null` and throws a runtime because of it, so you throw in an `if(species)` check to prevent it from runtiming.

Congrats, you fixed the runtime from showing up, but you didn't fix the larger problem of why species is null to begin with, which is the *real* problem.

So unless there's a specific reason why setting the vars on these radios isn't acceptable/doesn't work at all, this is papering it over, especially considering that the radio silencers *do* work a good portion of the time.

PidgeyThePirate commented:
As far as I can tell all the silencer does normally is turn the microphone and speakers off, which the microphone usually is anyway. Whether the microphone or speaker is on appears to have no bearing on whether or not someone can talk through it using Say. After testing I found that I could still talk through the headset perfectly fine once I got hit with the silencer, even though the microphone and speaker were off.

This is I believe the most elegant way to fix the issue as it uses a function that makes sense in the context of the silencer device itself, and as a bonus it keeps the speaker and microphone off after the EMP runs out, all without adding more code that's only used for that one interaction. You're against that, right?

Markolie commented:
Yup, I think this emp_act fix is the best solution as well as disabling the microphone doesn't actually prevent _usage_ of the microphone, it just stops it from broadcasting _everything_ in an X radius of being broadcast through it, similar to intercoms.

variableundefined commented on code/game/gamemodes/miniantags/abduction/abduction_gear.dm:
```diff
@@ -268,8 +268,9 @@
 			var/obj/item/radio/r = I
+			I.emp_act(1)
 			r.listening = 0
-			if(!istype(I, /obj/item/radio/headset))
```
EMP_act already set broadcasting to false

PidgeyThePirate commented:
Removed broadcasting=0 and changed the order of the remaining 2 lines so that listening = 0 comes first. That stops the radio from buzzing visibly to everyone in sight due to the EMP act.
